Lombok.NET/Extensions/SymbolExtensions.cs

```diff
@@ -50,4 +54,53 @@ public static string GetFullName(this ITypeSymbol typeSymbol)
 
 		return name;
 	}
-}
\ No newline at end of file
+
+	public static IEnumerable<ISymbol> GetAllMembersIncludingInherited(this ITypeSymbol typeSymbol)
+	{
+		return typeSymbol
+			.GetThisAndBaseTypes()
+			.SelectMany(t => t.GetMembers())
+			.Concat(typeSymbol.AllInterfaces.SelectMany(i => i.GetMembers()))
+			.Distinct(SymbolEqualityComparer.Default);
+	}
+
+	public static IEnumerable<ITypeSymbol> GetThisAndBaseTypes(this ITypeSymbol type)
+	{
+		var current = type;
+		while (current != null)
+		{
+			yield return current;
+			current = current.BaseType;
+		}
+	}
+
+	public static ParameterListSyntax GenerateParameterList(this ImmutableArray<IParameterSymbol> parameters) =>
+		ParameterList(SeparatedList(parameters.Select(p =>
+			Parameter(Identifier(p.Name))
+				.WithType(p.Type.ToTypeSyntax())
+				.WithModifiers(TokenList(p.RefKind.GenerateRefKindToken()))
+		)));
+
+	public static SyntaxToken GenerateAccessibilityToken(this IMethodSymbol symbol)
+	{
+		return symbol.DeclaredAccessibility switch
+		{
+			Accessibility.Public => Token(SyntaxKind.PublicKeyword),
+			Accessibility.Protected => Token(SyntaxKind.ProtectedKeyword),
+			Accessibility.Private => Token(SyntaxKind.PrivateKeyword),
+			_ => Token(SyntaxKind.None)
+		};
+	}
+
+	public static SyntaxToken GenerateRefKindToken(this RefKind refKind) => refKind switch
+	{
+		RefKind.Ref => Token(SyntaxKind.RefKeyword),
+		RefKind.Out => Token(SyntaxKind.OutKeyword),
+		RefKind.In => Token(SyntaxKind.InKeyword),
+		_ => Token(SyntaxKind.None)
+	};
+
+	public static TypeSyntax ToTypeSyntax(this ITypeSymbol type) =>
+		ParseTypeName(type.ToDisplayString()); //TODO Check if this works for generics
```

Comment:
A test for generics, and possibly something to do with ensuring the name is sufficiently qualified would be good here